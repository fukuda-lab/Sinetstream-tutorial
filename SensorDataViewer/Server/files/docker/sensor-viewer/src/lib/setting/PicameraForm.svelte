<script lang="ts">
  import { ValidationMessage } from "@felte/reporter-svelte";
</script>

<article class="border" aria-labelledby="picamera-group-title">
  <details open>
    <summary class="none">
      <div class="row">
        <h6 id="picamera-group-title">Raspberry Pi Camera</h6>
        <div class="max" />
        <i>arrow_drop_down</i>
      </div>
    </summary>

    <ValidationMessage for="image.topic" let:messages={message}>
      <div class="field label">
        <input type="text" name="image.topic" id="image-topic" />
        <label for="image-topic">image topic name</label>
        <span class="error">{message || ""}</span>
      </div>
    </ValidationMessage>

    <label class="switch icon">
      <input
        type="checkbox"
        name="image.visible"
        aria-label="image pane switch"
      />
      <span>
        <i>image</i>
      </span>
      <div class="tooltip">image pane</div>
    </label>

    <div class="field middle-align" style:margin-bottom="1rem">
      <label class="radio">
        <input
          type="radio"
          name="image.update"
          value="push"
          aria-label="radio push"
        />
        <span>push</span>
      </label>
      <label class="radio">
        <input
          type="radio"
          name="image.update"
          value="polling"
          aria-label="radio polling"
        />
        <span>polling</span>
      </label>
      <span class="helper">update method</span>
    </div>
  </details>
</article>

<style>
  .field.label > label {
    color: silver;
  }
</style>
