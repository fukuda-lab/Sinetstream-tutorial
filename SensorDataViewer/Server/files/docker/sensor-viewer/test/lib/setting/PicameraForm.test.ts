import { cleanup, render, screen } from '@testing-library/svelte';
import {
  afterEach, describe, expect, it,
} from 'vitest';
import PicameraForm from '../../../src/lib/setting/PicameraForm.svelte';

describe('PicameraForm.svelt', () => {
  afterEach(() => cleanup());

  it('mounts', () => {
    const { container } = render(PicameraForm);
    expect(container).toBeTruthy();
  });

  it('name', () => {
    render(PicameraForm);
    const item = screen.getByRole('textbox', { name: 'image topic name' });
    expect(item).toBeTruthy();
    expect(item.parentElement?.outerHTML).toMatchSnapshot();
    expect(item).toHaveProperty('name', 'image.topic');
    expect(item).toHaveProperty('type', 'text');
  });

  it('switch', () => {
    render(PicameraForm);
    const item = screen.getByRole('checkbox', { name: 'image pane switch' });
    expect(item).toBeTruthy();
    expect(item.parentElement?.outerHTML).toMatchSnapshot();
    expect(item).toHaveProperty('name', 'image.visible');
  });

  it('update method', () => {
    render(PicameraForm);
    const items = screen.getAllByRole('radio');
    expect(items).toBeTruthy();
    expect(items).toHaveLength(2);
    items.forEach((item) => {
      expect(item).toHaveProperty('name', 'image.update');
      expect(item.parentElement?.outerHTML).toMatchSnapshot();
    });
  });

  it('expansion open', () => {
    render(PicameraForm);
    const item = screen.getByRole('group');
    expect(item.getAttribute('open')).not.toBeNull();
  });
});
