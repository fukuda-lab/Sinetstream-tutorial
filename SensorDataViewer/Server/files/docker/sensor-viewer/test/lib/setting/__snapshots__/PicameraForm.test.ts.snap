// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PicameraForm.svelt > name 1`] = `"<div class=\\"field label s-w2M-6FE5mMvr\\"><input type=\\"text\\" name=\\"image.topic\\" id=\\"image-topic\\" class=\\"s-w2M-6FE5mMvr\\"> <label for=\\"image-topic\\" class=\\"s-w2M-6FE5mMvr\\">image topic name</label> <span class=\\"error s-w2M-6FE5mMvr\\"></span></div>"`;

exports[`PicameraForm.svelt > switch 1`] = `"<label class=\\"switch icon s-w2M-6FE5mMvr\\"><input type=\\"checkbox\\" name=\\"image.visible\\" aria-label=\\"image pane switch\\" class=\\"s-w2M-6FE5mMvr\\"> <span class=\\"s-w2M-6FE5mMvr\\"><i class=\\"s-w2M-6FE5mMvr\\">image</i></span> <div class=\\"tooltip s-w2M-6FE5mMvr\\">image pane</div></label>"`;

exports[`PicameraForm.svelt > update method 1`] = `"<label class=\\"radio s-w2M-6FE5mMvr\\"><input type=\\"radio\\" name=\\"image.update\\" value=\\"push\\" aria-label=\\"radio push\\" class=\\"s-w2M-6FE5mMvr\\"> <span class=\\"s-w2M-6FE5mMvr\\">push</span></label>"`;

exports[`PicameraForm.svelt > update method 2`] = `"<label class=\\"radio s-w2M-6FE5mMvr\\"><input type=\\"radio\\" name=\\"image.update\\" value=\\"polling\\" aria-label=\\"radio polling\\" class=\\"s-w2M-6FE5mMvr\\"> <span class=\\"s-w2M-6FE5mMvr\\">polling</span></label>"`;
